### PR TITLE
test: remove unused `norm_prv` parameter in `descriptor_tests.cpp`.

### PR DESCRIPTION
**Pull request by w0xlt**
This PR removes the unused `norm_prv` parameter in `src/test/descriptor_tests.cpp`.

### CODE REVIEW

**Comment by w0xlt** [comment]
It seems the original intention was to use `norm_prv` in the lines:

``` c++
std::string norm1;
BOOST_CHECK(parse_priv->ToNormalizedString(keys_priv, norma1));
BOOST_CHECK(EqualDescriptor(norm1, norm_pub));
```

But even descriptors that were parsed from an `xpriv` return the `xpub` in `ToNormalizedString`.
So this variable  `norm_prv`  is not being used anywhere.

File: src/test/descriptor_tests.cpp [hunk 1]
```diff
@@ -504,9 +505,9 @@ Check("sh(wsh(multi(20,KzoAz5CanayRKex3fSLQ2BwJpN7U52gZvxMyk78nDMHuqrUxuSJy,KwGN
...
+    Check("wsh(and_v(v:ripemd160(095ff41131e5946f3c85f79e44adbcf8e27e080e),multi(1,xprvA1RpRA33e1JQ7ifknakTFpgNXPmW2YvmhqLQYMmrj4xJXXWYpDPS3xz7iAxn8L39njGVyuoseXzU6rcxFLJ8HFsTjSyQbLYnMpCqE2VbFWc,xprv9uPDJpEQgRQfDcW7BkF7eTya6RPxXeJCqCJGHuCJ4GiRVLzkTXBAJMu2qaMWPrS7AANYqdq6vcBcBUdJCVVFceUvJFjaPdGZ2y9WACViL4L/0)))", "wsh(and_v(v:ripemd160(095ff41131e5946f3c85f79e44adbcf8e27e080e),multi(1,xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbmJbZRkrgZw4koxb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL,xpub68NZiKmJWnxxS6aaHmn81bvJeTESw724CRDs6HbuccFQN9Ku14VQrADWgqbhhTHBaohPX4CjNLf9fq9MYo6oDaPPLPxSb7gwQN3ih19Zm4Y/0)))", "wsh(and_v(v:ripemd160(095ff41131e5946f3c85f79e44adbcf8e27e080e),multi(1,xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbmJbZRkrgZw4koxb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL,xpub68NZiKmJWnxxS6aaHmn81bvJeTESw724CRDs6HbuccFQN9Ku14VQrADWgqbhhTHBaohPX4CjNLf9fq9MYo6oDaPPLPxSb7gwQN3ih19Zm4Y/0)))", UNSOLVABLE, {{"0020acf425291b98a1d7e0d4690139442abc289175be32ef1f75945e339924246d73"}}, OutputType::BECH32, {{},{0}});
+    // // Valid under sh(wsh()) and with a mix of xpubs and raw keys
```
Comment (darosior) [inline src/test/descriptor_tests.cpp:509]: Spurious change?

**Review by darosior — COMMENTED** [review]
Concept ACK. The checks against `norm_prv` were removed in commit 75530c93a83f3e94bcb78b6aa463c5570c1e737e (PR #21329) when the posisbility to serialize to normalize private string was removed. But the then-unused `norm_priv` wasn't removed.

File: src/test/descriptor_tests.cpp (hunk 1 above)
Comment (w0xlt) [reply src/test/descriptor_tests.cpp]: Done in https://github.com/bitcoin/bitcoin/pull/25863/commits/e5d77ce57dcf4fca1dc6ae34ff933f0f402e7068.

**Comment by w0xlt** [comment]
Rebased.

File: src/test/descriptor_tests.cpp [hunk 2]
```diff
@@ -373,60 +373,61 @@ BOOST_FIXTURE_TEST_SUITE(descriptor_tests, BasicTestingSetup)
...
-    Check("pk(5KYZdUEo39z3FPrtuX2QbbwGnNP5zTd7yyr2SC1j299sBCnWjss)", "pk(04a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd5b8dec5235a0fa8722476c7709c02559e3aa73aa03918ba2d492eea75abea235)", "pk(5KYZdUEo39z3FPrtuX2QbbwGnNP5zTd7yyr2SC1j299sBCnWjss)", "pk(04a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd5b8dec5235a0fa8722476c7709c02559e3aa73aa03918ba2d492eea75abea235)", SIGNABLE, {{"4104a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd5b8dec5235a0fa8722476c7709c02559e3aa73aa03918ba2d492eea75abea235ac"}}, std::nullopt);
-    Check("pkh(5KYZdUEo39z3FPrtuX2QbbwGnNP5zTd7yyr2SC1j299sBCnWjss)", "pkh(04a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd5b8dec5235a0fa8722476c7709c02559e3aa73aa03918ba2d492eea75abea235)", "pkh(5KYZdUEo39z3FPrtuX2QbbwGnNP5zTd7yyr2SC1j299sBCnWjss)", "pkh(04a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd5b8dec5235a0fa8722476c7709c02559e3aa73aa03918ba2d492eea75abea235)", SIGNABLE, {{"76a914b5bd079c4d57cc7fc28ecf8213a6b791625b818388ac"}}, OutputType::LEGACY);
+    // // Basic single-key uncompressed
```
Comment (theStack) [inline src/test/descriptor_tests.cpp:386]: another spurious change?
```suggestion
    // Basic single-key uncompressed
```

File: src/test/descriptor_tests.cpp [hunk 3]
```diff
@@ -373,60 +373,61 @@ BOOST_FIXTURE_TEST_SUITE(descriptor_tests, BasicTestingSetup)
...
+    // // Some unconventional single-key constructions
```
Comment (theStack) [inline src/test/descriptor_tests.cpp:394]:
```suggestion
    // Some unconventional single-key constructions
```

File: src/test/descriptor_tests.cpp [hunk 4]
```diff
@@ -373,60 +373,61 @@ BOOST_FIXTURE_TEST_SUITE(descriptor_tests, BasicTestingSetup)
...
+    Check("tr(03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5,{pk(03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5),{pk(L4rK1yDtCWekvXuE6oXD9jCYfFNV2cWRpVuPLBcCU2z8TrisoyY1),pk(03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5)}})", "tr(03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5,{pk(03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5),{pk(a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd),pk(03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5)}})", "tr(03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5,{pk(03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5),{pk(a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5bd),pk(03a34b99f22c790c4e36b2b3c2c35a36db06226e41c692fc82b8b56ac1c540c5)}})", XONLY_KEYS | SIGNABLE | MISSING_PRIVKEYS, {{"51201497ae16f30dacb88523ed9301bff17773b609e8a90518a3f96ea328a47d1500"}}, OutputType::BECH32M);
+
+    // // Versions with BIP32 derivations
```
Comment (theStack) [inline src/test/descriptor_tests.cpp:403]:
```suggestion
    // Versions with BIP32 derivations
```

File: src/test/descriptor_tests.cpp [hunk 5]
```diff
@@ -373,60 +373,61 @@ BOOST_FIXTURE_TEST_SUITE(descriptor_tests, BasicTestingSetup)
...
+    Check("pkh([01234567/10/20]xprv9s21ZrQH143K31xYSDQpPDxsXRTUcvj2iNHm5NUtrGiGG5e2DtALGdso3pGz6ssrdK4PFmM8NSpSBHNqPqm55Qn3LqFtT2emdEXVYsCzC2U/2147483647'/0)", "pkh([01234567/10/20]xpub661MyMwAqRbcFW31YEwpkMuc5THy2PSt5bDMsktWQcFF8syAmRUapSCGu8ED9W6oDMSgv6Zz8idoc4a6mr8BDzTJY47LJhkJ8UB7WEGuduB/2147483647'/0)", "pkh([01234567/10/20/2147483647']xpub69H7F5dQzmVd3vPuLKtcXJziMEQByuDidnX3YdwgtNsecY5HRGtAAQC5mXTt4dsv9RzyjgDjAQs9VGVV6ydYCHnprc9vvaA5YtqWyL6hyds/0)", HARDENED, {{"76a914ebdc90806a9c4356c1c88e42216611e1cb4c1c1788ac"}}, OutputType::LEGACY, {{10, 20, 0xFFFFFFFFUL, 0}});
+
+//     // Multisig constructions
```
Comment (theStack) [inline src/test/descriptor_tests.cpp:422]:
```suggestion
    // Multisig constructions
```

File: src/test/descriptor_tests.cpp [hunk 6]
```diff
@@ -449,9 +450,9 @@ Check("sh(wsh(multi(20,KzoAz5CanayRKex3fSLQ2BwJpN7U52gZvxMyk78nDMHuqrUxuSJy,KwGN
...
-    Check("sh(multi(2,[00000000/111'/222]xprvA1RpRA33e1JQ7ifknakTFpgNXPmW2YvmhqLQYMmrj4xJXXWYpDPS3xz7iAxn8L39njGVyuoseXzU6rcxFLJ8HFsTjSyQbLYnMpCqE2VbFWc,xprv9uPDJpEQgRQfDcW7BkF7eTya6RPxXeJCqCJGHuCJ4GiRVLzkTXBAJMu2qaMWPrS7AANYqdq6vcBcBUdJCVVFceUvJFjaPdGZ2y9WACViL4L/0))", "sh(multi(2,[00000000/111'/222]xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbmJbZRkrgZw4koxb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL,xpub68NZiKmJWnxxS6aaHmn81bvJeTESw724CRDs6HbuccFQN9Ku14VQrADWgqbhhTHBaohPX4CjNLf9fq9MYo6oDaPPLPxSb7gwQN3ih19Zm4Y/0))", "sh(multi(2,[00000000/111'/222]xprvA1RpRA33e1JQ7ifknakTFpgNXPmW2YvmhqLQYMmrj4xJXXWYpDPS3xz7iAxn8L39njGVyuoseXzU6rcxFLJ8HFsTjSyQbLYnMpCqE2VbFWc,xprv9uPDJpEQgRQfDcW7BkF7eTya6RPxXeJCqCJGHuCJ4GiRVLzkTXBAJMu2qaMWPrS7AANYqdq6vcBcBUdJCVVFceUvJFjaPdGZ2y9WACViL4L/0))", "sh(multi(2,[00000000/111'/222]xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbmJbZRkrgZw4koxb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL,xpub68NZiKmJWnxxS6aaHmn81bvJeTESw724CRDs6HbuccFQN9Ku14VQrADWgqbhhTHBaohPX4CjNLf9fq9MYo6oDaPPLPxSb7gwQN3ih19Zm4Y/0))", DEFAULT, {{"a91445a9a622a8b0a1269944be477640eedc447bbd8487"}}, OutputType::LEGACY, {{0x8000006FUL,222},{0}});
+//     // Checksums
```
Comment (theStack) [inline src/test/descriptor_tests.cpp:453]:
```suggestion
    // Checksums
```

**Review by theStack — COMMENTED** [review]
Concept ACK, good catch!

Seems like some comments have been changed unnecessarily, see below.

File: src/test/descriptor_tests.cpp (hunk 2 above)
Comment (w0xlt) [reply src/test/descriptor_tests.cpp]: Done in https://github.com/bitcoin/bitcoin/pull/25863/commits/57d1367fec7faec268b19d59fc1d6a98b2359de4.

File: src/test/descriptor_tests.cpp (hunk 3 above)
Comment (w0xlt) [reply src/test/descriptor_tests.cpp]: Done in https://github.com/bitcoin/bitcoin/commit/57d1367fec7faec268b19d59fc1d6a98b2359de4.

File: src/test/descriptor_tests.cpp (hunk 4 above)
Comment (w0xlt) [reply src/test/descriptor_tests.cpp]: Done in https://github.com/bitcoin/bitcoin/commit/57d1367fec7faec268b19d59fc1d6a98b2359de4.

File: src/test/descriptor_tests.cpp (hunk 5 above)
Comment (w0xlt) [reply src/test/descriptor_tests.cpp]: Done in https://github.com/bitcoin/bitcoin/commit/57d1367fec7faec268b19d59fc1d6a98b2359de4.

File: src/test/descriptor_tests.cpp (hunk 6 above)
Comment (w0xlt) [reply src/test/descriptor_tests.cpp]: Done in https://github.com/bitcoin/bitcoin/commit/57d1367fec7faec268b19d59fc1d6a98b2359de4.

**Comment by achow101** [comment]
ACK 57d1367fec7faec268b19d59fc1d6a98b2359de4

**Pull request merged**